src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -799,6 +800,12 @@ private void Parse()
             {
                 throw new MongoConfigurationException("This is an invalid w and journal pair.");
             }
+
+            string protectConnectionString(string connectionString)
+            {
+                var protectedString = Regex.Replace(connectionString, @"(?<=://)[^/]*(?=@)", "<hidden>");
```

Comment:
Kind of surprised we don't need to escape the forward slashes, but if the unit tests are passing then I guess we don't!